### PR TITLE
Release 2.9.4

### PR DESCRIPTION
**Pull request by poyzannur**
**What this PR does / why we need it**:
Updates changelog and release notes for 2.9.4 patch release

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] `CHANGELOG.md` updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,21 +1,21 @@
-## Unreleased
+## 2.9.4 (2024-01-24)
 
 ### All Changes
 
 #### Loki
 
 ##### Fixes
 
-* [11606](https://github.com/grafana/loki/pull/11606) **dannykopping** Fixed regression adding newlines to HTTP error response bodies which may break client integrations.
```
Comment (dannykopping) [inline CHANGELOG.md:9]: Curious why this was removed?

Comment (poyzannur) [reply CHANGELOG.md:9]: it didn't have the backport to 2.9.x tag, i deduced it wasn't backported? 
Let me double check if it's on the branch.

Comment (poyzannur) [reply CHANGELOG.md:9]: yep, it's not backported. would you like to include it in the patch release?

Comment (dannykopping) [reply CHANGELOG.md:9]: My bad! I _think_ this bug hadn't made it into 2.9.x which is why I didn't backport it
Lemme double check (and 2.8.x as well)

Comment (dannykopping) [reply CHANGELOG.md:9]: OK nvm, I was right, this bug never made it into the other releases 👍 please ignore

**Review by na-- — APPROVED** [review]

File: docs/sources/release-notes/v2-9.md [hunk 2]
```diff
@@ -49,3 +49,11 @@ For a full list of all changes and fixes, look at the [CHANGELOG](https://github
 * Fix querying ingester for label values with a matcher (previously didn't respect the matcher).
 * Ensure all lifecycler cfgs ref a valid IPv6 addr and port combination
 
+### 2.9.4 (2024-01-24)
```
Comment (DylanGuedes) [inline docs/sources/release-notes/v2-9.md:52]: nit: I know that we were already listing the different versions in increasing order (ex: first 2.9.1, then 2.9.2, etc), but I think the user experience is better if more recent releases are listed first. So 2.9.4, then 2.9.3, etc

Comment (na--) [reply docs/sources/release-notes/v2-9.md:52]: While I tend to agree, especially given the fact that in the 2.8.x branch the newest patch versions were first, see https://github.com/grafana/loki/blob/ned/release-v2.8.8/docs/sources/release-notes/v2-8.md :confused: 

I thought that someone had intentionally flipped the order for 2.9.x and so I didn't comment :sweat_smile:

Comment (poyzannur) [reply docs/sources/release-notes/v2-9.md]: thanks for catching, i actually didn't realised. 
i changed the order to be reverse.

**Pull request merged**